File: test/fixtures/python-sample/src/services/user_service.py
"""User service module."""

from datetime import datetime
import httpx

from src.models.user import User, UserCreate
from src.core.config import settings


# HACK: Using in-memory store instead of database
_users: list[User] = [
    User(id="1", name="John Doe", email="john@example.com", created_at=datetime.now()),
    User(id="2", name="Jane Smith", email="jane@example.com", created_at=datetime.now()),
]


class UserService:
    """Service class for user operations."""

    def __init__(self) -> None:
        """Initialize user service."""
        self.api_base_url = settings.EXTERNAL_API_URL

    async def get_all_users(self) -> list[User]:
        """Get all users."""
        return _users

    async def get_user_by_id(self, user_id: str) -> User | None:
        """Get a user by ID."""
        for user in _users:
            if user.id == user_id:
                return user
        return None

    async def create_user(self, user_data: UserCreate) -> User:
        """Create a new user."""
        new_user = User(
            id=str(len(_users) + 1),
            name=user_data.name,
            email=user_data.email,
            created_at=datetime.now(),
        )
        _users.append(new_user)
        return new_user

    async def fetch_external_user_data(self, user_id: str) -> dict:
        """Fetch user data from external API."""
        async with httpx.AsyncClient() as client:
            response = await client.get(f"{self.api_base_url}/users/{user_id}")
            response.raise_for_status()
            return response.json()

File: test/fixtures/typescript-sample/src/services/userService.ts
import axios from 'axios';

interface User {
  id: string;
  name: string;
  email: string;
  createdAt: Date;
}

interface CreateUserDto {
  name: string;
  email: string;
}

// TODO: Replace with actual database
const users: User[] = [
  { id: '1', name: 'John Doe', email: 'john@example.com', createdAt: new Date() },
  { id: '2', name: 'Jane Smith', email: 'jane@example.com', createdAt: new Date() },
];

export class UserService {
  private apiBaseUrl: string;

  constructor() {
    this.apiBaseUrl = process.env.API_BASE_URL || 'https://api.example.com';
  }

  async getAllUsers(): Promise<User[]> {
    // HACK: Using in-memory store instead of database
    return users;
  }

  async getUserById(id: string): Promise<User | undefined> {
    return users.find(user => user.id === id);
  }

  async createUser(dto: CreateUserDto): Promise<User> {
    const newUser: User = {
      id: String(users.length + 1),
      name: dto.name,
      email: dto.email,
      createdAt: new Date(),
    };
    users.push(newUser);
    return newUser;
  }

  // External API integration example
  async fetchExternalUserData(userId: string): Promise<any> {
    const response = await axios.get(`${this.apiBaseUrl}/users/${userId}`);
    return response.data;
  }
}
